hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AtttributeResolver.kt

```diff
@@ -91,22 +77,30 @@ private fun KProperty1<*, *>.getFieldType(): String? {
   return name
 }
 
+// The type of a property on the Prisoner record, used to derive searchable attributes
 private enum class PropertyType {
-  SIMPLE,
+  // A type that is supported because it has a dedicated matcher, e.g. DateTime
+  SUPPORTED_TYPE,
+
+  // An object type that won't have its own matcher, e.g. Address
   OBJECT,
-  LIST_SIMPLE,
-  LIST_OBJECT,
+
+  // A list of supported types - used only for validation to prevent people adding such lists to the Prisoner record
+  LIST_SUPPORTED_TYPES,
+
+  // A list of objects that won't have its own matcher, e.g. List<Address>
+  LIST_OBJECTS,
 }
 
-private fun getPropertyType(prop: KProperty1<*, *>): PropertyType =
-  when (prop.returnType.classifier) {
+private fun KProperty1<*, *>.getPropertyType(): PropertyType =
```

Comment:
We had a mixture of extension functions / KProperty passed as a parameter. Changed them all to extension functions for consistency.